test/datastore_test.py

```diff
@@ -1,61 +1,40 @@
-datastore management infrastructure breaks, we often don't find out about it
-for a long time. These tests are meant to help avoid that problem by
-continuosly exercising this functionality.
-"""
-
-import logging
-import os
-
-import pudl
-import pudl.workspace.datastore as datastore
-
-logger = logging.getLogger(__name__)
-
-
-def test_datastore(pudl_settings_fixture, data_scope):
-    """Download sample data for each available data source."""
-    sources_to_update = ['eia860', 'eia923', 'epaipm']
-    years_by_source = {
-        'eia860': data_scope['eia860_years'],
-        'eia923': data_scope['eia923_years'],
-        'epacems': [],
-        'epaipm': [None, ],
-        'ferc1': [],
-    }
-    # Sadly, FERC & EPA only provide access to their data via FTP, and it's
-    # not possible to use FTP from within the Travis CI environment:
-    if os.getenv('TRAVIS'):
-        states = []
-    else:
-        # Idaho has the least data of any CEMS state.
-        states = data_scope['epacems_states']
-        sources_to_update.extend(['ferc1', 'epacems'])
-        years_by_source['ferc1'] = data_scope['ferc1_years']
-        years_by_source['epacems'] = data_scope['epacems_years']
-
-    datastore.parallel_update(
-        sources=sources_to_update,
-        years_by_source=years_by_source,
-        states=states,
-        data_dir=pudl_settings_fixture['data_dir'],
-    )
-
-    pudl.helpers.verify_input_files(
-        ferc1_years=years_by_source['ferc1'],
-        epacems_years=years_by_source['epacems'],
-        epacems_states=states,
-        # Currently no mechanism for automatically verifying EPA IPM files...
-        pudl_settings=pudl_settings_fixture,
-    )
-    data_dir = pudl_settings_fixture['data_dir']
-    pudl.extract.eia860.Extractor(
-        data_dir).verify_years(years_by_source['eia860'])
-    pudl.extract.eia923.Extractor(
-        data_dir).verify_years(years_by_source['eia923'])
+"""Exercise the functionality in the datastore management module."""
+import random
+
+from pudl.workspace import datastore as datastore
+
+
+class TestDatastore:
+    """Test the Datastore functions."""
+
+    def test_doi_to_url(self, pudl_datastore_fixture):
+        """Get the DOI url right."""
+        number = random.randint(100000, 999999)  # nosec
+        fake_doi = f"10.5072/zenodo.{number}"
+
+        assert pudl_datastore_fixture.doi_to_url(fake_doi) == \
+            f"https://sandbox.zenodo.org/api/deposit/depositions/{number}"
+
+    def test_all_datapackage_json_available(self, pudl_datastore_fixture):
+        """
+        Ensure that every recorded DOI has a datapackage.json file.
+
+        Integration test!
+        """
+
+        def test_it(client, doi):
+            jsr = client.remote_datapackage_json(doi)
+
+            assert "title" in jsr, f"No title for {doi}"
+            assert "resources" in jsr, f"No resources for {doi}"
+            assert jsr["profile"] == "data-package", \
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/zaneselvans/pudl/pullRequest?prid=6263807)